crates/precompile/Cargo.toml

```diff
@@ -9,15 +9,18 @@ repository = "https://github.com/bluealloy/revm"
 version = "2.0.3"
 
 [dependencies]
-revm-primitives = { path = "../primitives", version="1.1.2", default-features = false }
+revm-primitives = { path = "../primitives", version = "1.1.2", default-features = false }
 bn = { package = "substrate-bn", version = "0.6", default-features = false }
 k256 = { version = "0.13", default-features = false, features = ["ecdsa"] }
 num = { version = "0.4.0", default-features = false, features = ["alloc"] }
-once_cell = "1.17"
 ripemd = { version = "0.1", default-features = false }
-secp256k1 = { version = "0.27.0", default-features = false, features = ["alloc", "recovery"], optional = true }
+secp256k1 = { version = "0.27.0", default-features = false, features = [
+    "alloc",
+    "recovery",
+], optional = true }
 sha2 = { version = "0.10.5", default-features = false }
 sha3 = { version = "0.10.7", default-features = false }
+lazy_static = "1.4"
```

Comment:
`lazy_static` is unmaintained. Please just use `once_cell` with `default-features = false`.

Reply:
That was my first try, doesn't work.

Without `std`, only `once_cell::unsync::OnceCell` is available, and it can't be used to initialize a static variable, because it is not `Sync`. It seems `lazy_static` manages to do it by using a spin-lock, what `once_cell` doesn't have.

Anyway, the other crate I removed, `to-binary`, is also unmaintained and is older, so I don't see the issue of using `lazy_static`, specially since neither `once_cell` or the unstable alternative `std::sync::LazyLock` is a suitable `Sync` no-std alternative.

Reply:
See the [`once_cell` FAQ](https://docs.rs/once_cell/latest/once_cell/#faq):

- Does this crate support `no_std`
- Should I use std::cell::OnceCell, once_cell, or lazy_static?

I say we enable the `critical-section` feature. cc @rakita

Reply:
Just found `once_cell::race`. I'll use it instead of `lazy_static`.

Reply:
OnceCell was stabilised inside rust, I would check if it works
https://doc.rust-lang.org/nightly/core/cell/struct.OnceCell.html#:~:text=Struct%20core%3A%3Acell%3A%3AOnceCell&text=A%20cell%20which%20can%20be,borrow%20checks%20(unlike%20RefCell%20).

to-binary is not relevant at all as it had very simpler usecase,

Reply:
It doesn't for static initialization, it is `!Sync`.

Reply:
> `core::sync::OnceLock` is the equivalent of `once_cell::sync::OnceCell`. Let's just use this and resolve this discussion :)

There is no `core::sync::OnceLock`. There is either `std::sync::OnceLock` or `core::cell::OnceCell`, the former requires `std`, the later is `!Sync`, thus it can't be a static variable.

Besides `lazy_static`, the valid alternatives are `once_cell::race::OnceCell` or `once_cell`'s `critical-section` feature. Given the revm usage, I think `once_cell::race::OnceCell` is a fine solution.

Reply:
> Sorry, it seems `once_cell::race` doesn't provide a generic `OnceCell`, instead it is specialized for some types that can be handled by atomic instructions. I'll change this PR to use critical-section.

Just came this to write this. We can use `once_cell::race::OnlyBox`, it does not require std and it fits nicely with the code

Reply:
Well, I just pushed a version using `critical-section`, had to add a new feature and all.

Reply:
> Well, I just pushed a version using `critical-section`, had to add a new feature and all. Should I change?

It would be better to change it, fewer features to maintain and it would work out of the box ( pun intended :))

Reply:
done